packages/protocol/contracts/identity/IdentityProxyHub.sol

```diff
@@ -0,0 +1,99 @@
+pragma solidity ^0.5.13;
+
+import "../common/Create2.sol";
+import "../common/UsingRegistry.sol";
+import "./IdentityProxy.sol";
+
+contract IdentityProxyHub is UsingRegistry {
+  bytes32 public constant identityProxyCodeHash = keccak256(
+    abi.encodePacked(type(IdentityProxy).creationCode)
+  );
+
+  mapping(bytes32 => IdentityProxy) public identityProxies;
+
+  /**
+   * @notice Returns the IdentityProxy address corresponding to the identifier.
+   * @param identifier The identifier whose proxy address is computed.
+   * @return The identifier's IdentityProxy address.
+   * @dev This function will return a correct address whether or not the
+   * corresponding IdentityProxy has been deployed. IdentityProxies are deployed
+   * using CREATE2 and this computes a CREATE2 address.
+   */
+  function getIdentityProxy(bytes32 identifier) public view returns (IdentityProxy) {
+    return IdentityProxy(Create2.computeAddress(address(this), identifier, identityProxyCodeHash));
+  }
+
+  /**
+   * @notice Returns the IdentityProxy address corresponding to the identifier,
+   * deploying an IdentityProxy if one hasn't already been deployed for this
+   * identifier.
+   * @param identifier The identifier whose proxy address is returned.
+   * @return The identifier's IdentityProxy address.
+   */
+  function getOrDeployIdentityProxy(bytes32 identifier) public returns (IdentityProxy) {
+    IdentityProxy identityProxy = identityProxies[identifier];
+    if (address(identityProxy) == address(0)) {
+      identityProxy = deployIdentityProxy(identifier);
+      identityProxies[identifier] = identityProxy;
+    }
+
+    return identityProxy;
+  }
+
+  /**
+   * @notice Returns true if the given address is the likely owner of the given
+   * identifier.
+   * @param addr The address to check.
+   * @param identifier The identifier to check.
+   * @return True if the given address is the likely owner of the given
+   * identifier, false otherwise.
+   */
+  function passesIdentityHeuristic(address addr, bytes32 identifier) public returns (bool) {
+    IAttestations attestations = getAttestations();
+    (uint32 completed, uint32 requested) = attestations.getAttestationStats(identifier, addr);
+
+    bool hasEnoughCompletions = completed >= 3;
```

Comment:
It would be nice to have the parameters of this heuristic be set as a contract field in the `constructor`. (Better still if it were `immutable` , but I notice we are not yet using Solidity ^0.6.5)

Reply:
Is it possible to set these as constants in a solidity constructor? Would be great to avoid unnecessary storage reads in the proxy functionality.

Reply:
So that's what [immutable](https://docs.soliditylang.org/en/v0.6.5/contracts.html#immutable) does IIUC. (i.e. immutable variables get writing into the deployed bytecode rather than being a storage variable)

Reply:
If we can't use `immutable` we might still use a `constant` just to make it a bit more parameterized (even though it can't be set in the constructor)